src/main/java/reposense/wizard/prompt/OptionalPrompt.java

```diff
@@ -0,0 +1,38 @@
+package reposense.wizard.prompt;
+
+import reposense.wizard.InputBuilder;
+
+/**
+ * Represents an optional prompt that only calls the provided prompt if requested by the user.
+ */
+public abstract class OptionalPrompt extends Prompt {
+    public static final String YES_FLAG = "Y";
+    public static final String NO_FLAG = "N";
+    private static final String FORMAT = String.format("%s/%s", YES_FLAG, NO_FLAG);
+
+    public OptionalPrompt(String description) {
+        super(description, FORMAT);
+    }
+
+    // optionally run array of Prompts if YES_FLAG is provided as input
+    public abstract Prompt[] optionallyRun();
+
+    @Override
+    public InputBuilder addToInput(InputBuilder inputBuilder) {
+        return inputBuilder;
+    }
+
+    @Override
+    public Prompt[] run() {
+        if (getResponse().compareToIgnoreCase(YES_FLAG) == 0) {
+            return optionallyRun();
+        }
+
+        if (getResponse().compareToIgnoreCase(NO_FLAG) == 0) {
+            return new Prompt[] {};
+        }
+
+        // Repeat OptionalPrompt until a valid input is provided
+        return new Prompt[] {this};
+    }
+}
```

Comment:
Not a requested change but more a point of discussion. Was wondering if there is a point in creating the subclasses since you could get this class to have a constructor that takes in the description and an instance of the prompt to be shown if Yes is chosen. 

I wrote this comment more from the perspective of testing since the subclass behaviors are almost identical I feel. See other comment about testing

Reply:
In what you mentioned, do you mean to have a single class per flag (e.g., SincePrompt should handle both asking if it should take in a SincePrompt, as well as actually taking in a SincePrompt)?

Reply:
In a previous implementation, it actually took in a `Prompt` in the constructor so we could avoid spawning optional subclasses. However, we decided that this design wass a little strange OOP-wise.

You can view the discussion me and @sikai00  had [here](https://github.com/sikai00/RepoSense/pull/2#discussion_r1133125714).

Reply:
I was referring to the version Marcus mentioned. I don't think cyclic dependency is correct for it, the dependency relations are almost the same in either case and is not cyclic since Prompt doesnt depend on OptionalPrompt. This just changes a dependency to an association. Regardless, I don't think it's a big issue